### PR TITLE
Solved issue 182

### LINKED ISSUE #182: Feat: Displaying Error Message from Backend in Toast Notification
### Describe the feature

Currently, the application makes an API call using Axios to create an account. On a successful response, a success toast message is shown, and the user is navigated to the home page. However, when an error occurs, a hard-coded error message (Failed To create account) is displayed in every case. I want to display the error message received from the backend in the toast notification instead of the hard-coded message.

### Add ScreenShots

Hardcoded

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Saksham2k3s**
This change addresses and fixes issue #182, which involves displaying the error message from the backend in a toast notification upon login and signup failure.

Fixes #182 

## Type of change

Please give a X on it which is applicable

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**_Test A Describe here_**

**_Test B Describe here (if Requred)_**

# Screenshorts and Vedios:

give screenshorts and vedio of the changes you made

# Checklist:
give a X on it which is applicable

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/pages/Login.jsx [hunk 1]
```diff
@@ -148,7 +118,7 @@ function Login() {
 
               <div className="relative mb-4">
                 <input
-                  required
```
Comment (Adityajparmar37) [inline src/pages/Login.jsx:151]: Why did you remove required ?? it is important to have password

File: src/pages/Signup.jsx [hunk 2]
```diff
@@ -165,7 +167,7 @@ function Signup() {
 
                 <div className="mb-4">
                   <input
-                    required
```
Comment (Adityajparmar37) [inline src/pages/Signup.jsx:168]: Here also why did you remove required

**Review by Adityajparmar37 — CHANGES_REQUESTED** [review]

**Comment by Saksham2k3s** [comment]
Sorry, it was a mistake. I was only testing some cases by removing this.

**Comment by Saksham2k3s** [comment]
I've added required in the input field

**Review by Adityajparmar37 — APPROVED** [review]

**Review by hustlerZzZ — APPROVED** [review]
Ok great work

**Pull request merged**